### ISSUE TITLE
[question] Download different package files into different path

### ISSUE THREAD

**Issue by Anchorli** [conan-io/conan#16912]
### What is your question?

Hello

I am using conan for different packages and different software development platform(SDP). I have two questions
The first one，I will depend on different packages when I build a firmware. But different packages have different download paths. How can I download different package into different paths. For example, I will download package A.h into include path and download package B.h into B/include path
The second one, different SDP will use same packages but the download path will be different. For example, one SDP download A.h into include/ path and another one download A.h into B/include path

To sum up，How can take over install phase when I use conan

### Have you read the CONTRIBUTING guide?

- [ ] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @Anchorli 

Thanks for your question.

It is not very clear what problem you are trying to solve, I think I didn't fully understand the question.

When packages are installed they are installed in different folders. Then, the generators like ``CMakeDeps`` create files that specify where the packages are so the build system can use them. Why this model is not working fine for your case?

If what you want is to "extract" the files from Conan packages and put them in a user-location, then what you are looking for is deployers, check for example this blog post: https://blog.conan.io/2023/05/23/Conan-agnostic-deploy-dependencies.html

There are both built-in deployers, but you can also write your deployer to do the copy and create the layout you want.

Please let us know if this is what you are looking for.

**Comment by memsharded** [comment]
Hi @Anchorli 

Any feedback for the above, or any further question? Thanks for the feedback.

**Comment by memsharded** [comment]
I am closing this ticket as responded, but please don't hesitate to open new tickets for any further question you might have. Thanks!
